### DISCUSSION TITLE
Issue loading Google Fonts to canvas

### DISCUSSION THREAD

**Discussion by ShuffledInk** [fabricjs/fabric.js#9060] [General]
I'm having an issue adding Google fonts to textbox objects. I have a select dropdown with fonts that should change the active object. It works with some of the fonts in my select menu, but not others. What is weird though, is if I select the fonts that don't work a second time, they will load, but the text selection gets thrown off.

Here is my code:

`const handleChange = (event) => {
  const selected = event.target.value;
    if (selected === "Times New Roman") {
      setFontFamily(selected);
      return;
    }
    const newFont = new FontFaceObserver(selected);
    newFont
      .load()
      .then(() => {
        console.log("loaded");
        canvas.getActiveObject().set({ fontFamily: selected });
        canvas.renderAll();
      })
      .catch((error) => {
        console.log(error);
      });
};`

Here is a preview of what's happening:

[screen-capture (9).webm](https://github.com/fabricjs/fabric.js/assets/125695464/26ed8c70-76bf-4589-98d7-f547801f7200)

In my preview, I selected a font that hasn't been working Aref Ruqaa. The first time I select it, it does not change. The second time I select it, it changes, but the text selection is way off. Same with the Lobster font. No console errors. When I log activeobject.get("fontFamily") the correct font is there.  Has anyone else run into a similar issue?

**Comment by ShaMan123** [comment]
Add a reproduction

**Reply by ShuffledInk** [reply]
I've managed to narrow down the problem. I realized this only happens when the text is added to the canvas with fontWeight: "bold" which leads me to believe it's something to do with the fonts themselves. I should be able to figure it out.

**Comment by ShuffledInk** [comment]
I figured it out. I needed to check in FontFaceObserver for bold.

` const newFont = new FontFaceObserver(selected, { weight: "bold" });`
